### ISSUE TITLE
VRRP module not getting the VRID information from an interface

### ISSUE THREAD

**Issue by davidban77** [arista-eosplus/pyeapi#170]
Hello,

I am using the `vrrp.get(<interface>).get(vrid)` method to collect VRRP group information but it seems  it is not retrieving any data. Below I am pasting what I am receiving along with the configuration on the switch:

**Swtich configuration**
```
interface Vlan177
   load-interval 5
   ip address 10.177.0.68/28
   vrrp 70 priority 200
   vrrp 70 ip 10.177.0.70
   vrrp 70 track ETH46 shutdown
   vrrp 71 priority 200
   vrrp 71 ip 10.177.0.71
   vrrp 71 shutdown
   vrrp 72 priority 200
   vrrp 72 ip 10.177.0.72
   vrrp 72 shutdown
```

**Python Example**
```
vrrp = node.api('vrrp')
>>> pprint(vrrp.get('Vlan177').get(71))
{'bfd_ip': '',
 'delay_reload': None,
 'description': '',
 'enable': True,
 'ip_version': None,
 'mac_addr_adv_interval': 30,
 'preempt': True,
 'preempt_delay_min': 0,
 'preempt_delay_reload': 0,
 'primary_ip': None,
 'priority': None,
 'secondary_ip': [],
 'timers_advertise': None,
 'track': []}
>>> pprint(vrrp.get('Vlan177').get(70))
{'bfd_ip': '',
 'delay_reload': None,
 'description': '',
 'enable': True,
 'ip_version': None,
 'mac_addr_adv_interval': 30,
 'preempt': True,
 'preempt_delay_min': 0,
 'preempt_delay_reload': 0,
 'primary_ip': None,
 'priority': None,
 'secondary_ip': [],
 'timers_advertise': None,
 'track': []}
```

You can see from the collected data that the `priority = None`, when the configuration is stated as 200, and even though the `vrrp 71 shutdown` it returns it as `enabled = True`.

Was this working before? should I use another method?

Thanks,

**Comment by Alexanderis** [comment]
Hi David,

Could you let me know the following:
- EOS version you are testing against
- pyeapi version you are using

I tested on 4.20.11 code and it looks fine on both pyeapi (0.6.1) and pyeapi (0.8.2).
```
import pyeapi
from pprint import pprint as pp
node = pyeapi.connect_to('172.28.161.116')
node.enable('show hostname')
vrrp = node.api('vrrp')
pp(vrrp.get('Vlan177').get(70))
pp(vrrp.get('Vlan177').get(71))
pp(vrrp.get('Vlan177').get(72))
```
Outputs look good including all details like IPs and priority.
```
{'bfd_ip': '',
 'delay_reload': 0,
 'description': '',
 'enable': True,
 'ip_version': 2,
 'mac_addr_adv_interval': 30,
 'preempt': True,
 'preempt_delay_min': 0,
 'preempt_delay_reload': 0,
 'primary_ip': '10.177.0.70',
 'priority': 200,
 'secondary_ip': [],
 'timers_advertise': 1,
 'track': [{'action': 'shutdown', 'name': 'ETH46'}]}
{'bfd_ip': '',
 'delay_reload': 0,
 'description': '',
 'enable': False,
 'ip_version': 2,
 'mac_addr_adv_interval': 30,
 'preempt': True,
 'preempt_delay_min': 0,
 'preempt_delay_reload': 0,
 'primary_ip': '10.177.0.71',
 'priority': 200,
 'secondary_ip': [],
 'timers_advertise': 1,
 'track': []}
{'bfd_ip': '',
 'delay_reload': 0,
 'description': '',
 'enable': False,
 'ip_version': 2,
 'mac_addr_adv_interval': 30,
 'preempt': True,
 'preempt_delay_min': 0,
 'preempt_delay_reload': 0,
 'primary_ip': '10.177.0.72',
 'priority': 200,
 'secondary_ip': [],
 'timers_advertise': 1,
 'track': []}
```

**Comment by davidban77** [comment]
Hi Alexandru,

Sure:
- `pyeapi=0.8.2`
- `4.21.3F`

I am testing on the installed python of the switch just to test a simple script. The python installed is 2.7.15

I am creating a container to test it with with Python 3

**Comment by davidban77** [comment]
I thought that some data parsing different on the python version, but I still do not get the data of the VRRP group

```
Python 3.7.2 (default, Feb  6 2019, 01:47:18)
[GCC 8.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyeapi
>>> node = pyeapi.connect_to('localhost')
>>> vrrp = node.api('vrrp')
>>> from pprint import pprint
>>> pprint(vrrp.get('Vlan177').get(70))
{'bfd_ip': '',
 'delay_reload': None,
 'description': '',
 'enable': True,
 'ip_version': None,
 'mac_addr_adv_interval': 30,
 'preempt': True,
 'preempt_delay_min': 0,
 'preempt_delay_reload': 0,
 'primary_ip': None,
 'priority': None,
 'secondary_ip': [],
 'timers_advertise': None,
 'track': []}
>>> pprint(node.enable('show vrrp interface vlan177'))
[{'command': 'show vrrp interface vlan177',
  'encoding': 'json',
  'result': {'virtualRouters': [{'bfdPeerAddr': '0.0.0.0',
                                 'description': '',
                                 'groupId': 70,
                                 'interface': 'Vlan177',
                                 'macAddressInterval': 30,
                                 'masterAddr': '10.177.0.68',
                                 'masterDownInterval': 3210,
                                 'masterInterval': 1,
                                 'masterPriority': 200,
                                 'preempt': True,
                                 'preemptDelay': 0,
                                 'preemptReload': 0,
                                 'priority': 200,
                                 'skewTime': 0.21,
                                 'state': 'master',
                                 'trackedObjects': [{'interface': 'Ethernet46',
                                                     'priority': 0,
                                                     'shutdown': True,
                                                     'trackedObject': 'ETH46'}],
                                 'version': 2,
                                 'virtualIp': '10.177.0.70',
                                 'virtualIpSecondary': [],
                                 'virtualMac': '00:00:5e:00:01:46',
                                 'vrIdDisabled': False,
                                 'vrIdDisabledReason': '',
                                 'vrfName': 'default',
                                 'vrrpAdvertInterval': 1}]}}]
```

**Comment by Alexanderis** [comment]
Thanks for the details.
Let me dig into this and will let you know once I have more info.

**Comment by Alexanderis** [comment]
Hi David,

Apologies for the delayed reply. Looks like there is an issue with 4.21.3F version which is causing VRRP module to break. This should be fixed in EOS 4.21.4 version at which time we'll also update the client.

Sorry for any inconvenience caused.

Regards,
Alex.

**Comment by davidban77** [comment]
Thanks for the update Alex, I can workaround using operational commands for the moment.

So I'll close this one now.

Cheers,
David
